interfaces/cython/cantera/ck2yaml.py

```diff
@@ -691,13 +691,14 @@ def __init__(self, label, geometry, well_depth, collision_diameter,
                  dipole_moment, polarizability, z_rot, note=''):
 
         try:
-            geometry = int(geometry)
-        except ValueError:
-            raise InputError(
-                "Bad geometry flag '{}' for species '{}', is the flag a float "
-                "or character? It should be an integer.", geometry, label)
-        if geometry not in (0, 1, 2):
-            raise InputError("Bad geometry flag '{}' for species '{}'.", geometry, label)
+            if float(geometry) == int(float(geometry)):
+                geometry = int(float(geometry))
```

Comment:
The original `int` conversion should remain in the `try` branch, with the current check being moved to the `except` branch. A warning should be issued so that the conversion does not remain silent.

Reply:
I thought about doing the nested `try`/`except` block and keeping the `int` conversion in the original `try` branch, but I supposed that anything that failed `int(geometry)` would also fail `float(geometry)` (in the case of a random string) since the former would be stricter anyway. I also thought it would be interesting to jsut `raise InputError(...) from` whatever `ValueError` occurred, whether the two that were explicitly raised or the one that might occur from `float(geometry)`. 

But I'm more than happy to change it into form suggested! I don't know how pull requests normally work, so do I just make the changes on my branch, make a new commit, and then include them in the pull request somehow?

Reply:
> I don't know how pull requests normally work, so do I just make the changes on my branch, make a new commit, and then include them in the pull request somehow?

Yes, make a new commit then `git push` to the same branch. That will automatically update this pull request with the new commits.

> And I certainly agree on the warning, rather than silently fixing it. Is there an existing custom warning type that should be used for input syntax issues, or is the standard warnings.warn preferred?

I think the best place to look would be to do whatever is done for duplicate thermo entries. At the least, this warning would be behind the `--permissive` flag, and an error if `--permissive` is not given.

Reply:
@ischoegl ... sorry, was a bit delayed by the hurricane but I believe I figured everything out 😅

@bryanwweber I changed the test case for floats (since that is now accepted behavior) to verify that it fails for floating point rounding errors instead (e.g. `1.999999999999999999`)

Let me know if there's anything else that needs to be changed or added.

Reply:
@cory-kinney see below :point_down: I think you need to change the test code as well to make sure that things are failing and succeeding when they're supposed to.